ChronoKeeper/DataCollectionService.h

```diff
@@ -0,0 +1,84 @@
+#ifndef DATA_COLLECTION_SERVICE_H
+#define DATA_COLLECTION_SERVICE_H
+
+#include <iostream>
+#include <margo.h>
+#include <thallium.hpp>
+#include <thallium/serialization/stl/string.hpp>
+
+#include "chronolog_types.h"
+#include "KeeperDataStore.h"
+
+namespace tl = thallium;
+
+namespace chronolog
+{
+
+class DataCollectionService : public tl::provider<DataCollectionService> 
+{
+public:
+    
+// Service should be created on the heap not the stack thus the constructor is private...
+
+    static DataCollectionService * CreateDataCollectionService(tl::engine& tl_engine, uint16_t service_provider_id
+		    , KeeperDataStore & dataStoreInstance)
+    {
+          return  new DataCollectionService( tl_engine, service_provider_id, dataStoreInstance );
+    }  
+
+    ~DataCollectionService() 
+    {
+	//remove provider finalization callback from the engine's list	
+        get_engine().pop_finalize_callback(this);
+    }
+
+    void collection_service_available(tl::request const& request)
+    {   
+	    request.respond(1);
+    }
+
+    void shutdown_data_collection(tl::request const& request)
+    {   
+	    int status =1;
+	    theDataStore.shutdownDataCollection();
+	    request.respond(1); 
+    }
+
+    void StartStoryRecording(tl::request const& request, 
+                      std::string const& chronicle_name, std::string const& story_name,  StoryId const& story_id)
+    {
+        std::cout << "DataCollectionService: StartStoryRecoding {"<< story_name<<":"<<story_id<<"}"<< std::endl;
+        theDataStore.startStoryRecording(chronicle_name, story_name, story_id);
+        request.respond( story_id );
```

Comment:
I believe this will be called after Client's AcquireStory request reaches Visor. Do you think this should be called synchronously before AcquireStory returns? Or do you think we can do this in the background?